### PR TITLE
Make container create, start, and stop events consistent

### PR DESCRIPTION
**Pull request by smarterclayton**
The messages for container lifecycle events are subtly inconsistent and should be unified.

First, the field format for containers (`spec.containers{installer}`) is hard to parse for a human,
so include the container name directly in the message for create and start to be consistent with kill.

Second, the pulling image event has inconsistent capitalization, fix that to be sentence without punctuation.

Third, the kill container event was unnecessarily wordy and inconsistent with the create and start events. Make the following changes:

* Use 'Stopping' instead of 'Killing' since kill is usually reserved for when we decide to hard stop a container
* Send the event before we dispatch the prestop hook, since this is an "in-progress" style event vs a "already completed" type event
* Remove the 'cri-o://' / 'docker://' prefix by printing the container name instead of id (we already do that replacement at the lower level to prevent high cardinality events)
* Use 'message' instead of 'reason' as the argument name since this is a string for humans field, not a string for machines field
* Remove the hash values on the container spec changed event because no human will ever be able to do anything with the hash value
* Use 'Stopping container %s(, explanation)?' form without periods to follow event conventions

The end result is a more pleasant message for humans:

```
35m         Normal    Created                       Pod    Created container
35m         Normal    Started                       Pod    Started container
10m         Normal    Killing                       Pod    Killing container cri-o://installer:Need to kill Pod
10m         Normal    Pulling                       Pod    pulling image "registry.svc.ci.openshift.org/openshift/origin-v4.0-2019-02-10-172026@sha256:3da5303d4384d24691721c1cf2333584ba60e8f82c9e782f593623ce8f83ddc5"
```

becomes

```
35m         Normal    Created                       Pod    Created container installer
35m         Normal    Started                       Pod    Started container installer
10m         Normal    Killing                       Pod    Stopping container installer
10m         Normal    Pulling                       Pod    Pulling image "registry.svc.ci.openshift.org/openshift/origin-v4.0-2019-02-10-172026@sha256:3da5303d4384d24691721c1cf2333584ba60e8f82c9e782f593623ce8f83ddc5"
```

/kind bug
/kind cleanup

```release-note
Events reported for container creation, start, and stop now report the container name in the message and are more consistently formatted.
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
@kubernetes/sig-node-pr-reviews 

@derekwaynecarr addressing some egregious inconsistencies among container lifecycle events - the container name in the message reads ok even in big lists (better than `fieldPath: spec.containers{installer}` which is a bit baroque), while pod name is under involvedObject in both new and old schema.

**Comment by smarterclayton** [comment]
I may also update the "Kind" column to be a calculated field including the name if set (namespace is usually irrelevant).

**Comment by smarterclayton** [comment]
Changed the `Kind` column in https://github.com/kubernetes/kubernetes/pull/73894 to improve locating the object

**Review by feiskyer — COMMENTED** [review]
It's looking much better. LGTM.

Also, add a release note?

**Comment by smarterclayton** [comment]
Updated release note

/retest

**Comment by smarterclayton** [comment]
Applying tag based on lgtm
